### PR TITLE
Partially butchered creatures can't be resurrected by laz injectors

### PR DESCRIPTION
**Pull request by Sprok0**
Creatures that have been butchered can no longer by revived by lazarus injectors. This is mostly to keep a miner from easily maxing out their hardsuit after killing only one goliath.

The basilisk diamonds and hivelord core are converted into butchery drops, rather than just appearing when the mob dies.

Fixes a minor bug where a butchered mob's sprite was updated before the butchering completed, mostly visible with spider sprites having one more leg than they actually had left.

Butchering products can have the time it takes to butcher them changed, rather than being the same for all of them.

Goliath hides now give 20 melee resistance to mining suits rather than 10, needing three to reach the maximum protection of 90.

### CODE REVIEW

**Comment by Intigracy** [comment]
`>The basilisk diamonds and hivelord core are converted into butchery drops, rather than just appearing when the mob dies.`
`>combined with the change above it`
See attached emoji with my name on it.

**Comment by aceedex** [comment]
Nah, that's gay as fuck tbh.

Specially since to get maxed gear, you need to kill twelve (12) goliaths

That's 12,000 dosh just on Lazarus capsules alone. There are twenty of them in the miningvend overall, without counting extra lazarous to keep the mob alive as a pet, 2000 more for accelerators and more for balls

Also, you need to spend the time to kill and revive all 12.  That's time where you're not mining and not making the aforementioned money. It's hard enough to find 12 goliaths in your average round, now imagine competing with two other miners

**Comment by clusterfack** [comment]
12 goliaths? What changed? In my day it was 5 for a max 90% reduction.

**Comment by Sprok0** [comment]
It's five for the suit and five for the helmet, I don't know where he got 12 though.
I'll make each goliath hide give 20 res instead of 10 if not being able to grind out the perfect suit is really such an issue

**Comment by aceedex** [comment]
Oh yeah, I'm being dumb and was counting for 100% resistance, but it caps at 90.

That still doesn't change the fact encountering ten goliaths is a fucking pipe dream. If there were a lot more I'd understand, but Mining being my 2nd most played job, I can tell you that I've only maxed out a goliath belt once.  

I'd say this works as it is. It's fairly expensive so you need to at least do one delivery to sci ( a timely delivery of 15 of each ore usually yielding about 3000 cash money) and then go to town with a D-drill. And THEN spend an assload of time killing the goliath, skinning them, reviving them, waiting for the accelerators to charge and such.

It just works, and there's no reason to change it. 90 melee res is really not all it's cracked up to be, and there's really no way to get it before the 30 minute mark without first doing a really good job of mining  AND spending time to actually do it

**Comment by Sprok0** [comment]
Now it's only 3 to max out each piece, 2 if you're ok with stopping at 80.

**Comment by aceedex** [comment]
Or you could leave a system that works well enough alone

**Comment by coldcola** [comment]
This seems wholly unnecessary taken at the goliath level, MAYBE for other creatures if people were abusing this to get infinite amounts of a specific meat, but as it stand laz injectors are too expensive for that to be considered by most.

Your argument for this merge loses a lot of momentum focusing on the goliath hardsuits - which I rarely if ever see.

**Comment by Sprok0** [comment]
Does anyone actually buy the injectors? I usually just see the vending machines spitting them out for free to anyone nearby

**Comment by coldcola** [comment]
Does anyone actually mine? I usually just see them at the bar or are in it for the free hardsuits

**Comment by invoid9** [comment]
If a dude gets the money for injectors, finds an exotic creature and spends all of his money on those injectors to get """""infinite""""" meat, How is that a bad thing? The person spent a lot more money for a large supply of a specific meat, when he could easily get pretty much infinite synth meat from chem/botany. Pay more, get more. Unless I am missing an actual exploit.

(Same applies for diamonds from Basilisks)

**Comment by Sprok0** [comment]
Meat isn't a problem, it's building up 90 melee resist on the mining hardsuit with goliath armor plates. The idea was that building up armor should be a reward for finding and killing goliaths, not killing one and hacking the vender/using some of your otherwise worthless money.
I added the basilisk diamonds/hivelord core to the butchering list because I thought they would make more sense there rather than just popping out on death.

**Comment by Shadowmech88** [comment]
> The basilisk diamonds and hivelord core are converted into butchery drops, rather than just appearing when the mob dies.

This makes sense, as far as the game's internal logic goes.

That being said, it also prevents you from using a hivelord as a reusable 1000 credit full heal. That's not a fun change and I hate it.

**Comment by Sprok0** [comment]
It's still a 1000 credit full heal, just only once per hivelord. You could also drag around a hivelord corpse around in a locker or something and take out the core if you need it, the timer only starts when the core is actually spawned.

**Comment by Shadowmech88** [comment]
> just only once per hivelord

Yes, this is exactly the part I don't like.
I'm completely biased, but knowing that doesn't make me dislike the change any less.

**Comment by coldcola** [comment]
> Meat isn't a problem, it's building up 90 melee resist on the mining hardsuit with goliath armor plates. 

it's fucking nothing

> I added the basilisk diamonds/hivelord core to the butchering list because I thought they would make more sense there rather than just popping out on death.

this should have been it, you've gone too far for no reason.

**Comment by Sprok0** [comment]
near immunity to melee attacks isn't nothing, what are you talking about
I just spent like three minutes beating someone with a fully upgraded suit using a toolbox, they died of blood loss from a lucky hit that got through and caused a bleed, and died with 95 brute damage after literally more than 80 toolbox whacks to the chest.
Same deal with an energy sword, 68 continuous hits to the chest before they went into crit, again from bloodloss after lucky hits get through.
90 melee protection is _extremely _strong__

**Comment by pvak** [comment]
i died to x pls nerf

**Comment by SonixApache** [comment]
Good change, fuck lazarus abuse and creature farming.

**Comment by aceedex** [comment]
No, fam. In our stun-centric battle system where you can slip, wire restrain someone  take off armor and kill 90 melee res is fucking nothing 

There is literally nothing wrong with a system that let's you be completely invulnerable to the worst possible attack method after you put in 40 or so mimutes of grinding into it

**Comment by coldcola** [comment]
@SonixApache 

> Good change, fuck lazarus abuse and creature farming.

Can you explain how you see this feature as abusive? The way I'm seeing it the time and credit investment is enough to make this a moot request, other than the butchery changes which make sense.

This like the lockbox PR negatively affects more good well-intentioned staff than anyone else who might potentially 'abuse' it.

**Comment by FudgePucker** [comment]
I disagree with this PR.
:-2:*100

**Comment by Sprok0** [comment]
One plasma ore is is worth 40 credits. Each plasma mineral tile drops 5 ore. Digging and refining 50 plasma tiles, which are very common, is enough to buy out a vending machine's supply of lazarus injectors. At that point, you go and sit somewhere on the asteroid for 40 minutes beating a non-hostile goliath until your suit is powered up. This is not a "feature" worth keeping, it is an exploit in how the lazarus injector revive function works.

**Comment by aceedex** [comment]
exploit? Well, ignoring the stupid argument of what is and isn't an exploit

1. Mining 40 plasma veins with a pick is really fucking time consuming. On top of that, you'd also have to take into account actually grinding the plates. Complete waste of time, when again, if you really needed that 90 melee resistance you'd just buy a riot suit
2. If you mine it with a d-drill, the congratulations, you had to spend 15 or so minutes on the asteroid actually doing mining so you could deliver a good haul so that science could give you a drill. Now all you have to do is spend ages killing those goliaths for a sightly better riot suit

What does this PR even fix? what's the issue? fixing "exploits" for the sake of fixing is stupid if it takes out something that well, people seem to like to an extent, is perfectly balanced, and has really no negative impacts in the game itself
what is the point?

And on another issue, you'd make healing with a hivelord take a lot longer since you'd have to wait and skin it. That's fucking great. I just got mauled by a goliath and I'm bleeding out but now I have to sit here and wait while I bleed out because someone though an extremely cumbersome and expensive method of healing was too strong or something?

**Comment by Sprok0** [comment]
The time it takes to butcher a specific butchering product is now configurable, removing spider/frog legs takes half the default time, removing a hivelord core takes a tenth.

**Comment by SonixApache** [comment]
@coldcola you can carry a full heal with you wherever you go -- which is bullshit. And you get to re-use it for only 1000 credits. Which is even worse.

@aceedex exploits should be fixed -- regardless of people who think they're positive because they abuse them the most.

**Comment by aceedex** [comment]
Assumptions are a terrible argument, and that  entire post of yours is as well. 
Not like there's anything to exploit. having to do a boring repetitive task so you could have spaceproof riot gear at 13:00 is hardly something worth doing, but the option is nice.

**Comment by Sprok0** [comment]
Its more than just a full heal, it's the same proc that changelings use to revive. It fully restores blood, replaces any missing limbs, etc

**Comment by coldcola** [comment]
I have absolutely no problem with miners having a cheap  insta-heal if they go through the motions to get it, they're most most exposed and easy to kill profession in the game.

The way you argue for your change and make adjustments makes me wonder if again, you've been affected negatively by this and took it upon yourself to make a fix for something that isn't a bug.

**Comment by SonixApache** [comment]
> they're most most exposed and easy to kill profession in the game.

free 90 brute protection hardsuits, enough money to buy anything (including their job-specific 15 brute energy guns) and access to the cargo computers to use that money, an entire Z-level that you only share with up to 5 others TOPS, and re-usable adminorazine-tier rejuvenators aren't exactly what I'd call exposed or easy to kill. 

Having to kill an easy-to-kill mob and then walk into some walls for a minute or two isn't nearly enough effort to get something as strong as a re-usable rejuvenator.

And no, the usual "He was killed by x so now he's crying to remove it" meme doesn't apply to me -- never has. 
This is an exploit. It's equal to killing a human, eating their heart, then defibbing them to continue from step one. Where do they get the extra hearts?

**Comment by coldcola** [comment]
> free 90 brute protection hardsuits, enough money to buy anything (including their job-specific 15 brute energy guns) and access to the cargo computers to use that money, an entire Z-level that you only share with up to 5 others TOPS, and re-usable adminorazine-tier rejuvenators aren't exactly what I'd call exposed or easy to kill.

none of these are going to stop a changeling who parastings you, or an antag that has you as a target and managed to procure some means of stunning you momentarily.  it's fucking nothing.

**Comment by aceedex** [comment]
> free 90 brute protection hardsuits, enough money to buy anything (including their job-specific 15 brute energy guns)
The fact that you'd call an hour of work "free" or even pretend that money has any sort of use, OR that you'd pretend the protokinetic accelerator is superior to anything but ghetto weapons really makes me wonder. 

Remember, whoever stuns first wins. A hivelord is fucking nothing if you don't have time to throw the ball, kill it and then eat it. Maybe it's good in a place where there's tons of simple mobs that seldom stun you and is easy to get away from but can still cause severe damage

Hm. Wonder where that could be

**Comment by SonixApache** [comment]
Yes, not much apart from voice activated flashbang / chloralsmoke implants will save you from the evil evil stun. Yes, the surprise element is usually the most important thing in any battle. But let's save that for another discussion.

Okay, so maybe the hardsuits don't immediately have 90 brute protection, but they've got decent armor to start with. And goliaths are as common as there are slots on your mining to fit PKAs in. Which, coincidentally, kill a goliath in one mag and maybe a whack or two with your mining tool.

Money gets you food, chems, powerful weapons and armor, materials, tools... all without even leaving your department. Only thing it doesn't get you is a hardsuit, but you've got that already.

PKAs do 30 brute per shot (at <50kPa) and cause internal bleeding. I think they get about 8 shots on a full charge? If you manage to hit half of that, your opponent's pretty much done for. It's not the captain's laser, but it's not a bad weapon in the slightest.

And yes, if you don't have time to hide for a second, whack a thing 4-6 times and click on yourself, your pocket rejuvenator won't be that effective. Doesn't take away that it's a ridiculously powerful exploit. especially if you've got any kind of support already.

_Long story short; this PR fixes an exploit, and that's a good thing. Always._

**Comment by PJB3005** [comment]
Hang on a fucking second are people actually complaining that this bug fix is a bad thing?

Jesus fuck get your shit together.

**Comment by aceedex** [comment]
Goliaths are not that common. I've had rounds where I've found 2 at best, and that's with me actually going out to look for them. Somtimes there's less, and sometimes there's more. Extremely uncertain to find the 6 that you need to for a maxed out suit. It's making something that's worthless even more worthless

@PJB3005 no you

**Comment by coldcola** [comment]
Probably because it's more of a fix to an oversight, not really a bug,

**Comment by PJB3005** [comment]
This is a bug. Bugs need fixing.

If you feel that without this bug it's not balanced correctly, that's a separate matter entirely and for a separate discussion.

**Comment by aceedex** [comment]
So if this gets merged, and I make a PR that reverts it and call it a feature, It'll get merged because features are good right?

File: code/datums/helper_datums/butchering.dm [hunk 1]
```diff
@@ -185,11 +193,24 @@
...
+
+/datum/butchering_product/hivelord_core
+	result = /obj/item/asteroid/hivelord_core
+	verb_name = "remove the core from"
+	verb_gerund = "removing the core from"
+	butcher_time = 2
+
+/datum/butchering_product/hivelord_core/desc_modifier()
+	if(!amount)
+		return "Its core has been taken"
```
Comment (Intigracy) [inline code/datums/helper_datums/butchering.dm]: .

**Comment by coldcola** [comment]
@PJB3005 Yes I saw the bug tag you added, I still disagree with the idea that the revivification of butchered simple mobs with the lazarus injectors is a bug, more of an oversight.  
How else are you to revive the station pets when some asshole removes their legs??

everything else he touched was an oversight or bug though, the butchering menu, the icon refresh etc.

**Comment by clusterfack** [comment]
If there is a way to get multiple goliath hides etc from the same goliath, it needs to be fixed, simple as that.

**Comment by Sprok0** [comment]
Gonna work on a better way to do this

**Comment by ghost** [comment]
How does the mob keep track of how much it has been butchered?

Does this carry through with rejuv?

If so, correct that

**Comment by FudgePucker** [comment]
90 brute isn't shit when you still go down in one stun.
What is the first thing they are going to do to you after stunning you?
Stripping you of your exosuit to remove your jumpsuit to rid of your sensors.
So instead of killing you first, they just strip you of your hardsuit and let space help them kill you.

No really, anyone who thinks that there is even nearly enough goliaths on the asteroid to even get two plates of armor has never played mining. It is a good mob spawn for the round if there is a goliath even on the asteroid.
Do half of the coders wanting this oversight to get fixed even play on the server anymore?

**Comment by ghost** [comment]
Last I checked, 90 brute basically made you immune to mob attacks, which is the intended effect of reinforcing the mining hardsuit.

And, despite the tentacles, I've never seen a goliath strip a miner after stunning them

**Comment by SonixApache** [comment]
I've found at least two goliaths every time I've mined in the past two weeks.

**Comment by aceedex** [comment]
Goliaths are still quite a bit of a threat even with a maxed suit
reason being, they can still shatter bones and cause bleeding, as most other attacks with melee res

**Pull request closed without merging**